### ISSUE TITLE
Design: UI to add reusable Q&As to Campaign Landing Pages (CLPs)

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#14443]
### User story

**AS AN** Editor inexperienced in Drupal
**I WANT** to be able to add reusable Q&As in addition to custom/one-off Q&As
**SO THAT** I can draw on the pre-existing Q&A library if it's helpful, while still creating my own campaign-specific Q&As if needed outside of the Q&A library.

### Notes / background

In planning for Sprint 89, we (including Dave C) realized that we don't have a clear enough handle on how reusable Q&As should exist alongside the existing FAQ section. 

Drupal UI design should flow from answers to these Veteran-facing UX questions:
- Should reused Q&As be separate from FAQs? (e.g., a "related questions" section or something)
- If reused Q&As are _not_ separate, should they be kept together, i.e., appended to the end of the FAQ (vs. mixed in with one-off ones)

Editor-facing questions that came up during #10941:
- See [this Mural](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1689892596393/dc147aedc3e6182dbe0ad911a4c10877ac06c129?sender=uebde62ceb342c6477fb09406) for observations about how naming conventions and help text around Q&As might need to change in order to make them less confusing for CLP Editors.
  - Note that the names (Q&A, Q&A Group, Q&A Section, etc?) are used in multiple content types, e.g. Benefit Detail Page uses 3 of them(!)
  
### Design

[Drupal UI design in Figma](https://www.figma.com/file/MTBtolRaZK4hUkCprXyIiD/Q%26As?type=design&node-id=4%3A2157&mode=design&t=J07jlBp0p9rkSrAl-1)

## Acceptance criteria

- [x] ~~Veteran-facing UX defined in Sketch~~ Jordan determined this won't be needed. See more detailed comment below. 
- [x] Drupal UI defined in Figma ([link to Figma file](https://www.figma.com/file/MTBtolRaZK4hUkCprXyIiD/Q%26As?type=design&node-id=4%3A2157&mode=design&t=J07jlBp0p9rkSrAl-1))
  - [x] Naming conventions and help text clarified
- [ ] Reviewed by product manager
- [x] Reviewed by a11y
- [ ] Reviewed by PO

**Comment by chri5tia** [comment]
Not sure if this is helpful, but I see an opportunity to add it here, so when they click to add the Q&A, they can chose if it's a single (stand alone) or a Q&A group (reusable). This is how it works in other places.

![Screenshot 2023-07-20 at 2 38 01 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/22764938/4a510d1d-3661-462c-8ee5-96c604dc8f89)

From a backend engineering stand-point, it wouldn't be a heavy lift to do it that way.

Also, what do you think about us changing "Q&A - Single" to something to the affect of standalone and changing "Q&A Group" to something to the affect of reusable?

Currently when you click on the button to add a Q&A it opens the form to add a single not resuable one:
![Screenshot 2023-07-20 at 2 42 11 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/22764938/dd34ee24-1752-4b55-9137-f370586e0dd3)

In other places, where they have a choice, it pops up as a modal to search for what you want to add:
<img width="545" alt="Screenshot 2023-07-20 at 3 20 29 PM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/22764938/5afacea9-d459-4044-b579-9d863bcfa35e">

**Comment by chri5tia** [comment]
Some background:

Q&A - Single: one question and it's answer, that can be resuable
Q&A Group: one Q&A - Single, chosen from a bank
Q&A - a question and answer, not reusable
Q&A Section - a group of Q&As (not reusable)

**Comment by wesrowe** [comment]
@thejordanwood, note that I added some notes and a Mural link (regarding naming conventions and help text re "Q&A things") to the ticket.

**Comment by wesrowe** [comment]
UX refinement – Jordan may opt to continue her investigation of the Editor experience of Q&As beyond CLPs as a follow-on to this ticket. I.e., look at naming conventions and help text in other content types, both for reusables and one-offs.

Content types where reusables have already been added:
- R&S pages (done)
- Benefit Details (in process) - this one will have both Q&A Group and Q&A Section available in the "add content block" UI

I guess Jordan has the option to do the beyond-CLP work above as part of this ticket, since any changes to naming conventions would affect those Editor experiences...? I can go either way.

**Comment by jilladams** [comment]
Per https://dsva.slack.com/archives/C0FQSS30V/p1690313218806759 moving out of sprint, to stretch.

**Comment by thejordanwood** [comment]
I've created a [design in Figma](https://www.figma.com/file/MTBtolRaZK4hUkCprXyIiD/Q%26As?type=design&node-id=4%3A2157&mode=design&t=J07jlBp0p9rkSrAl-1) that shows suggested updates to the Drupal UI.

**Comment by thejordanwood** [comment]
**End of sprint update:**

I've created a Figma document that shows the Drupal UI and plan to create the Front end Veteran-facing UX in Sketch next sprint.

**Comment by thejordanwood** [comment]
Thoughts on FE design of CLPs:

I’m thinking reusable Q&As should go on the bottom of the FAQ section. This is assuming that we think the custom/single Q&As will be more tailored to the page content and will end up being more relevant to the user, which would be the reasoning that they should appear at the top.
If we add them to the bottom, then I don’t see the FE actually changing design. It’ll still look like this.

![Screenshot 2023-08-02 at 4 36 08 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/104526584/057f7f0a-6e18-44dd-b689-48d7cff65919)


Also if we add them to the bottom, this will make the [CMS design I have mocked up](https://www.figma.com/file/MTBtolRaZK4hUkCprXyIiD/Q%26As?type=design&node-id=0%3A1&mode=design&t=HZicYFAUynIFnqJJ-1) make more visual sense. I have the reusable/existing Q&As below the custom/single Q&As.

![Screenshot 2023-08-02 at 4 40 22 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/104526584/7a2f2bd3-5460-4820-94c3-931d73d44735)

[Slack conversation about this topic](https://dsva.slack.com/archives/C52CL1PKQ/p1691012483996269)

**Comment by thejordanwood** [comment]
In [my Figma file](https://www.figma.com/file/MTBtolRaZK4hUkCprXyIiD/Q%26As?type=design&node-id=0%3A1&mode=design&t=HZicYFAUynIFnqJJ-1) I also showed what the Drupal UI could be for Benefit Detail Pages and Resource & Support pages to make the way we use/refer to reusable Q&As consistent.  

Some thoughts on the FE of those pages:

- Benefit Detail Pages: Order of Q&As is determined by the order they’re placed in the Main Content section.
- Resources & Support page: FE design will stay the same because I think this page only supports reusable Q&As?

**Comment by DanielleThierryUSDSVA** [comment]
Hi @wesrowe. Sure, of course, happy to explain this.

Many CLPs are attached to campaigns that have information that is also core benefit information. So, for example, there might be a CLP focused on the new VA Life insurance program. The CLP is framing this as a new program and maybe has videos and what not to highlight that.

But the communications team might also want to include some FAQs about what the program is and how it works. This is also core benefit information.

**- What we have now:** Communications teams create custom FAQs, repeating information from the core benefit information. Content team tries to keep up with chasing down various iterations of the same content. But we cannot scale as a human COPE machine :)

**- What we want:** Communications team create their CLP with their intro, videos, etc. And then they pull in reusable Q&As so we are always presenting consistent, accurate information.

Make sense?

**Comment by wesrowe** [comment]
That is super helpful, @DanielleThierryUSDSVA, thank you!

**Comment by thejordanwood** [comment]
[Slack thread](https://dsva.slack.com/archives/C52CL1PKQ/p1691081655635009) with a11y and design feedback.

**Comment by wesrowe** [comment]
@thejordanwood, I see this is in the Review column. What are next steps from your perspective?

**Comment by thejordanwood** [comment]
@wesrowe Does this need sign off from anyone else? If not, I think the next step would be getting this into development.

**Comment by thejordanwood** [comment]
Closing this ticket. Design for the UI has been made. I'm going to review with Drupal engineers next sprint.

**Comment by wesrowe** [comment]
Ticket for reviewing with engineers and studying impact across all content types with Q&As: #14790.

**Comment by wesrowe** [comment]
@thejordanwood, we got feedback from Erika (CMS CC) that it's preferred we don't carry forward the word "Section" in the Q&A UI at all. I asked whether that's a "must" or just "should/could."

Meantime, I noticed that you only use the word "section" isn't used a lot in your Figma designs. And I wondered, did you talk about calling that concept "Group" and did Danielle and Amanda think that was viable?
